crates/viewer/re_dataframe_ui/src/filters/string.rs

```diff
@@ -0,0 +1,299 @@
+use std::any::Any;
+use std::fmt::Formatter;
+use std::sync::Arc;
+
+use arrow::array::{
+    ArrayRef, BooleanArray, Datum, LargeStringArray, ListArray, StringArray, StringViewArray,
+    as_list_array,
+};
+use arrow::datatypes::{DataType, Field};
+use datafusion::common::{Column, Result as DataFusionResult, exec_err};
+use datafusion::logical_expr::{
+    ArrayFunctionArgument, ArrayFunctionSignature, ColumnarValue, Expr, ScalarFunctionArgs,
+    ScalarUDF, ScalarUDFImpl, Signature, TypeSignature, Volatility, col, lit,
+};
+
+use re_ui::SyntaxHighlighting;
+use re_ui::syntax_highlighting::SyntaxHighlightedBuilder;
+
+use super::{FilterUiAction, action_from_text_edit_response};
+
+#[derive(Debug, Clone, Copy, Default, PartialEq, Eq)]
+pub enum StringOperator {
+    #[default]
+    Contains,
+    StartsWith,
+    EndsWith,
+}
+
+impl std::fmt::Display for StringOperator {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Contains => "contains".fmt(f),
+            Self::StartsWith => "starts with".fmt(f),
+            Self::EndsWith => "ends with".fmt(f),
+        }
+    }
+}
+
+impl StringOperator {
+    pub const ALL: &'static [Self] = &[Self::Contains, Self::StartsWith, Self::EndsWith];
```

Comment:
we use a lot of those around here. Makes me nervous they outdate. Have you checked if we have some minimal procmacro around for this? I know we use `enumset` in some places, but that one seems like overkill. Almost tempted to roll our own just for the `ALL` property